### PR TITLE
Support for HashAgg Function

### PR DESCRIPTION
**Pull request by Rajeshwari4**

### CODE REVIEW

File: core/src/main/java/org/apache/calcite/sql/fun/SqlLibraryOperators.java [hunk 1]
```diff
@@ -1714,4 +1714,7 @@ private SqlLibraryOperators() {
   public static final SqlAggFunction MEDIAN =
       new SqlMedianAggFunction(SqlKind.MEDIAN, ReturnTypes.ARG0_NULLABLE);
 
+  @LibraryOperator(libraries = {SNOWFLAKE})
```
Comment (tejaswan55) [inline core/src/main/java/org/apache/calcite/sql/fun/SqlLibraryOperators.java:1717]: test cases?

Comment (Rajeshwari4) [reply core/src/main/java/org/apache/calcite/sql/fun/SqlLibraryOperators.java:1731]: Added TCs.

**Comment by nosheenSheikh02** [comment]
Add test case for to_json_string separately.

**Comment by nosheenSheikh02** [comment]
Please resolve conflicts

File: core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java [hunk 2]
```diff
@@ -12415,6 +12493,52 @@ RelNode createLogicalValueRel(RexNode col1, RexNode col2) {
...
+        .project(builder.alias(bitXor, "value"))
+        .build();
+
+    final String expectedBiqQuery = "SELECT BIT_XOR(FARM_FINGERPRINT(TO_JSON_STRING(SAL))) AS value\n"
+        + "FROM scott.EMP";
+
+    assertThat(toSql(root, DatabaseProduct.BIG_QUERY.getDialect()), isLinux(expectedBiqQuery));
+  }
+
+  @Test public void testForJsonStringFunction() {
```
Comment (nosheenSheikh02) [inline core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java:12527]: can u pls change this to 
testForToJsonStringFunction

as it is denoting the exact function name and existing one do not

Comment (Rajeshwari4) [reply core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java]: Done

**Review by dm-shantanuc — APPROVED** [review]
LGTM

**Pull request merged**